repondeur/zam_repondeur/models/lecture.py

```diff
@@ -19,22 +22,52 @@ class Lecture(Base):  # type: ignore
     chambre = Column(Text, primary_key=True)
     session = Column(Text, primary_key=True)
     num_texte = Column(Integer, primary_key=True)
+    organe = Column(Text, primary_key=True)
     titre = Column(Text)
 
-    @property
-    def chambre_disp(self) -> str:
+    def __str__(self) -> str:
+        return ", ".join(
+            [
+                self.format_chambre(),
+                self.format_session(),
+                self.format_organe(),
+                self.format_texte(),
+            ]
+        )
+
+    def format_chambre(self) -> str:
         return CHAMBRES[self.chambre]
 
-    def __str__(self) -> str:
-        return f"{self.chambre_disp}, session {self.session}, texte nº {self.num_texte}"
+    def format_session(self) -> str:
+        if self.chambre == "an":
+            return f"{self.session}e législature"
+        else:
+            return f"session {self.session}"
+
+    def format_organe(self) -> str:
+        result: str = self.organe
+        organes = get_data("organes")
+        if self.organe in organes:
+            organe_data = organes[self.organe]
+            result = organe_data["libelleAbrege"]
+        return self.rewrite_organe(result)
+
+    def rewrite_organe(self, label: str) -> str:
```

Comment:
I'm not sure if that's pertinent but there is that [particular line](https://github.com/betagouv/zam/blob/master/repondeur/zam_repondeur/fetch.py#L53) that might reuse it?

Reply:
Hmm, I think the logic in that line might be wrong.

If the case of the Assemblée nationale, the text we want to get the articles of might be:
- the text as initially submitted → `f"{BASE_URL}{lecture.session}/projets/pl{lecture.num_texte:04}.asp"`
- the text as modified by a commission → `f"{BASE_URL}leg/pjl{lecture.session[2:4]}-{lecture.num_texte:03}.html"`

The URL pattern does not depend on whether the lecture is commission or séance publique, it depends on what happened before, and we don't keep that information handy... 😞

Reply:
Maybe the easy way is to just try both URLs and see which one works...

Reply:
Ok, so I did a separate PR for this (#67), tell me what you think.